### PR TITLE
[test] Extend DGEMM test for using MKL

### PR DESCRIPTION
**Pull request by lucamar**
I have added the build and run of `DGEMM` using `PrgEnv-intel` with `MKL`. 
The performance values are still the same, I have tested `Dom` for the time being and the performance fits the current range. 
I have inserted a new tag `benchmark`, therefore the performance thresholds might change with respect to `diagnostic`.

### CODE REVIEW

**Comment by lucamar** [comment]
Note that you need to run `reframe -P normal` on `Dom` to get the node assignment correctly.

File: cscs-checks/libraries/math/dgemm.py [hunk 1]
```diff
@@ -33,9 +33,26 @@ def __init__(self):
         }
 
         self.maintainers = ['AJ', 'VH', 'VK']
-        self.tags = {'diagnostic'}
+        self.tags = {'benchmark', 'diagnostic'}
 
     def setup(self, partition, environ, **job_opts):
+
+        if environ.name.startswith('PrgEnv-gnu'):
+            envname = 'PrgEnv-gnu'
```
Comment (victorusu) [inline cscs-checks/libraries/math/dgemm.py]: As far as I can see, the `envname` variable is defined but never used.

File: cscs-checks/libraries/math/dgemm.py [hunk 2]
```diff
@@ -33,9 +33,26 @@ def __init__(self):
...
+            envname = 'PrgEnv-intel'
+            self.build_system.cflags += [
+                '-qopenmp', '-DMKL_ILP64', '-I${MKLROOT}/include',
+                '-Wl,--start-group',
+                '${MKLROOT}/lib/intel64/libmkl_intel_ilp64.a',
+                '${MKLROOT}/lib/intel64/libmkl_intel_thread.a',
+                '${MKLROOT}/lib/intel64/libmkl_core.a',
+                '-Wl,--end-group',
+                '-liomp5', '-lpthread', '-lm', '-ldl']
+        else:
```
Comment (victorusu) [inline cscs-checks/libraries/math/dgemm.py]: I think you can remove completely this `else:` branch.

File: cscs-checks/libraries/math/dgemm.py (hunk 1 above)
Comment (lucamar) [reply cscs-checks/libraries/math/dgemm.py]: Is it used if I select on the command line `-p PrgEnv-gnu` when I invoke `reframe`?

File: cscs-checks/libraries/math/dgemm.py [hunk 3]
```diff
@@ -33,9 +33,22 @@ def __init__(self):
...
+            self.build_system.cflags += ['-fopenmp']
+        elif environ.name.startswith('PrgEnv-intel'):
+            self.build_system.cflags += [
+                '-qopenmp', '-DMKL_ILP64', '-I${MKLROOT}/include',
+                '-Wl,--start-group',
+                '${MKLROOT}/lib/intel64/libmkl_intel_ilp64.a',
+                '${MKLROOT}/lib/intel64/libmkl_intel_thread.a',
+                '${MKLROOT}/lib/intel64/libmkl_core.a',
+                '-Wl,--end-group',
+                '-liomp5', '-lpthread', '-lm', '-ldl']
```
Comment (vkarak) [inline cscs-checks/libraries/math/dgemm.py]: To be typically correct the options should be as follows:

```python
self.build_system.cppflags = ['-DMKL_ILP64', '-I{MKLROOT}/include']
self.build_system.cflags = ['-qopenmp']
self.build_system.ldflags = ['--the-rest']
```

Comment (lucamar) [reply cscs-checks/libraries/math/dgemm.py]: Thanks @vkarak, I have modified the code as you suggested

**Comment by vkarak** [comment]
@victorusu Can you recheck and approve?

**Comment by vkarak** [comment]
I will move the test to microbenchmarks

**Comment by victorusu** [comment]
I will merge, as soon as the tests pass.

**Pull request merged**